app/src/org/commcare/utils/AndroidUtil.java

```diff
@@ -49,29 +49,23 @@ public static int generateViewId() {
      * Initialize platform specific methods for common handlers
      */
     public static void initializeStaticHandlers() {
-        DataUtil.setUnionLambda(new AndroidUnionLambda());
+        DataUtil.setIntersectionLambda(new AndroidIntersectionLambda());
     }
 
-    private static class AndroidUnionLambda extends UnionLambda {
+    public static class AndroidIntersectionLambda extends IntersectionLambda {
 
         @Override
-        public <T> Vector<T> union(Vector<T> a, Vector<T> b) {
+        public <T> Vector<T> intersection(Vector<T> a, Vector<T> b) {
             Vector<T> result = new Vector<>();
```

Comment:
Can you move initializing this to right before it's used? Will help readability a bit